### ISSUE TITLE
True SSL in video_jepa?

### ISSUE THREAD

**Issue by Hardwarize** [facebookresearch/eb_jepa#14]
Hi team,
I have a question regarding the Moving MNIST training loop. In the provided code, the total loss is calculated as:
`total_loss = jepa_loss + recon_loss + det_loss`

Since `det_loss `uses the ground truth digit locations (loc_map) and is part of the `total_loss.backward()` step, does this mean the JEPA encoder is being trained with supervised signals? 

In a "pure" JEPA, at least as I undertand it, the representation is usually learned only through self-supervised prediction and regularization. I wanted to clarify,  just to make sure that I've undertood well the paradigm,  if this joint training is an intentional design choice for this library or if the `detection_head ` should technically be trained as a frozen probe to maintain a purely self-supervised encoder.

Thanks!

**Comment by montoneguglielmo** [comment]
I had a similar question and looked deeper into the code and this is my understanding. I hope this helps! :)

The short answer is: no, the JEPA encoder does not receive supervised gradient signals from det_loss.

Here's why:

The JEPAProbe wrapper blocks gradients
Both pixel_decoder and detection_head are instances of JEPAProbe 
[jepa.py:208-237](https://github.com/facebookresearch/eb_jepa/blob/main/eb_jepa/jepa.py#L208)
which wraps the encoder with a double detachment:

```
def forward(self, observations, targets):
    with torch.no_grad():                  # no gradient computation through encoder
        state = self.jepa.encode(observations)
    output = self.head(state.detach())     # extra detach for safety
    return self.hcost(output, targets)

```
So even though the training loop writes:
total_loss = jepa_loss + recon_loss + det_loss
total_loss.backward()

the recon_loss and det_loss gradients are completely stopped at the probe head boundary. 
They never reach the encoder.

**Comment by Hardwarize** [comment]
Thanks @montoneguglielmo! yes it makes sense. I thought that I've not seen previously this online gradient stop method.
